packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -453,6 +460,29 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
           columns: visibleColumns,
         });
       }
+
+      // If the selected row is not within the current range of rows being displayed,
+      // we need to render it at either the top or bottom of the rows,
+      // depending on whether it is above or below the range.
+      if (!isFocusedCellRendered && cell && !disableVirtualization) {
+        const rows = currentPage.rows.filter((row) => row.id === cell.id);
+        const focusedRow = rows.length > 0 && rows[0];
+
+        if (focusedRow) {
+          const index = currentPage.rows.indexOf(focusedRow);
+          if (index > firstRowToRender) {
```

Comment:
```suggestion
        const focusedRowIndex = currentPage.rows.findIndex((row) => row.id === cell.id);
        const focusedRow = currentPage.rows[focusedRowIndex];

        if (focusedRow) {
          if (focusedRowIndex > firstRowToRender) {
```

Use `.findIndex` instead of `.filter` + `.indexOf`?

Reply:
done.I have also moved it out of the function so it won't have to find the index every time if the focused cell is not changed